### PR TITLE
doc: Document undocumented environment variables

### LINKED ISSUE #500: Exposure of undocumented environment variables
There are undocumented environment variables, which are useful for adjustment, troubleshooting, and so on.

## Overview

- `EC_ENDPOINT` & `EC_HOST` for [endpoint](https://registry.terraform.io/providers/elastic/ec/latest/docs#endpoint')
- `EC_INSECURE` & `EC_SKIP_TLS_VALIDATION` for [insecure](https://registry.terraform.io/providers/elastic/ec/latest/docs#insecure)
- `EC_TIMEOUT` for [timeout](https://registry.terraform.io/providers/elastic/ec/latest/docs#timeout)
- `EC_VERBOSE` for [verbose](https://registry.terraform.io/providers/elastic/ec/latest/docs#verbose)
- `EC_VERBOSE_CREDENTIALS` for [verbose_credentials](https://registry.terraform.io/providers/elastic/ec/latest/docs#verbose_credentials)
- `EC_VERBOSE_FILE` for [verbose_file](https://registry.terraform.io/providers/elastic/ec/latest/docs#verbose_file)

## Possible Implementation

Insert the following sentence before the part of `Conflicts with...` / `Not recommended...` in each.

- It can also be sourced from the `EC_XXX` environment variable.
- It can also be sourced from the `EC_XXX` or `EC_YYY` environment variables

## Testing

n/a because of documentation.

## Context

Without knowing `EC_TIMEOUT`, I may have to embed the timeout setting to do `terraform apply` using WiFi in the airplane.

## Your Environment

```
Terraform v1.2.2
on linux_amd64
+ provider registry.terraform.io/elastic/ec v0.5.0
```

### PR DESCRIPTION
**Pull request by sakurai-youhei**
Adds these in documentation.
- EC_ENDPOINT & EC_HOST
- EC_INSECURE & EC_SKIP_TLS_VALIDATION
- EC_TIMEOUT
- EC_VERBOSE
- EC_VERBOSE_CREDENTIALS
- EC_VERBOSE_FILE

Closes #500

This PR mines undocumented environment variables from ec provider code [HERE](https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L84-L161).

## Description

Here are origins of each.

- EC_ENDPOINT & EC_HOST: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L90
- EC_INSECURE & EC_SKIP_TLS_VALIDATION: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L126
- EC_TIMEOUT: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L135
- EC_VERBOSE: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L143
- EC_VERBOSE_CREDENTIALS: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L151
- EC_VERBOSE_FILE: https://github.com/elastic/terraform-provider-ec/blob/master/ec/provider.go#L159

## Related Issues

#500

## Motivation and Context

For example, switching `EC_VERBOSE` is very useful when encountering intermit and temporary failures on CI servers.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
 
 * `endpoint` - (Optional) This is the target endpoint. It must be provided only when
    you use the Elastic Cloud provider with an ECE installation or ESS Private.
+   It can also be sourced from the `EC_ENDPOINT` or `EC_HOST` environment variables.
```
Comment (nrichers) [inline docs/index.md:84]:
```suggestion
  Can also be sourced from the `EC_ENDPOINT` or `EC_HOST` environment variables.
```

File: docs/index.md [hunk 2]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
   `apikey`. Not recommended.
 
 * `insecure` - (Optional) This setting allows the provider to skip TLS verification.
-  Useful when targeting installation with self-signed certificates. Not recommended when
+  Useful when targeting installation with self-signed certificates. It can also be sourced from the
```
Comment (nrichers) [inline docs/index.md:99]:
```suggestion
  Useful when targeting installation with self-signed certificates. Can also be sourced from the
```

File: docs/index.md [hunk 3]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
+  `EC_INSECURE` or `EC_SKIP_TLS_VALIDATION` environment variables. Not recommended when
   targeting ESS.
 
 * `timeout` - (Optional) This setting allows the user to set a custom timeout in the
   individual HTTP request level. Defaults to 1 minute (`"1m"`), but might need to be tweaked if timeouts
-  are experienced.
+  are experienced. It can also be sourced from the `EC_TIMEOUT` environment variable.
```
Comment (nrichers) [inline docs/index.md:105]:
```suggestion
  are experienced. Can also be sourced from the `EC_TIMEOUT` environment variable.
```

File: docs/index.md [hunk 4]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 
 * `verbose` - (Optional) When set to `true`, it writes a `requests.json` file in the folder
   where Terraform runs with all the outgoing HTTP requests and responses. Defaults to `false`.
+  It can also be sourced from the `EC_VERBOSE` environment variable.
```
Comment (nrichers) [inline docs/index.md:109]:
```suggestion
  Can also be sourced from the `EC_VERBOSE` environment variable.
```

File: docs/index.md [hunk 5]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 
 * `verbose_credentials` - (Optional) When set with `verbose`, the contents of the Authorization
-header will not be redacted. Defaults to `false`.
+  header will not be redacted. Defaults to `false`. It can also be sourced from the
```
Comment (nrichers) [inline docs/index.md:112]:
```suggestion
  header will not be redacted. Defaults to `false`. Can also be sourced from the
```

File: docs/index.md [hunk 6]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
+  `EC_VERBOSE_CREDENTIALS` environment variable.
 
 * `verbose_file` - (Optional) Sets the file where the verbose request and response HTTP flow will
-be written to. Defaults to `request.log`.
+  be written to. Defaults to `request.log`. It can also be sourced from the `EC_VERBOSE_FILE`
```
Comment (nrichers) [inline docs/index.md:116]:
```suggestion
  be written to. Defaults to `request.log`. Can also be sourced from the `EC_VERBOSE_FILE`
```

**Review by nrichers — COMMENTED** [review]
Some minor edits you can commit directly. I would just omit the "it" in reference content, especially if you are referring to a parameter from earlier sentences.

**Comment by sakurai-youhei** [comment]
@nrichers Thanks for the heads up on that point. I was also uncomfortable with those "it" but I wasn't sure what can make those right/better, so that helps much. :)

**Review by tobio — COMMENTED** [review]
IMO it would be nicer to say 'Can be overridden by...' so the precedence is clear.

File: docs/index.md [hunk 7]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
   targeting ESS.
 
 * `timeout` - (Optional) This setting allows the user to set a custom timeout in the
   individual HTTP request level. Defaults to 1 minute (`"1m"`), but might need to be tweaked if timeouts
-  are experienced.
+  are experienced. Can also be sourced from the `EC_TIMEOUT` environment variable.
 
 * `verbose` - (Optional) When set to `true`, it writes a `requests.json` file in the folder
   where Terraform runs with all the outgoing HTTP requests and responses. Defaults to `false`.
+  Can also be sourced from the `EC_VERBOSE` environment variable.
```
Comment (sakurai-youhei) [inline docs/index.md:109]:
```suggestion
  where Terraform runs with all the outgoing HTTP requests and responses. Can also be sourced
  from the `EC_VERBOSE` environment variable. Defaults to `false`.
```

File: docs/index.md [hunk 8]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 
 * `verbose_credentials` - (Optional) When set with `verbose`, the contents of the Authorization
-header will not be redacted. Defaults to `false`.
+  header will not be redacted. Defaults to `false`. Can also be sourced from the
+  `EC_VERBOSE_CREDENTIALS` environment variable.
```
Comment (sakurai-youhei) [inline docs/index.md:113]:
```suggestion
  header will not be redacted. Can also be sourced from the `EC_VERBOSE_CREDENTIALS`
  environment variable. Defaults to `false`.
```

File: docs/index.md [hunk 9]
```diff
@@ -95,18 +96,22 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 
 * `verbose_file` - (Optional) Sets the file where the verbose request and response HTTP flow will
-be written to. Defaults to `request.log`.
+  be written to. Defaults to `request.log`. Can also be sourced from the `EC_VERBOSE_FILE`
+  environment variable.
```
Comment (sakurai-youhei) [inline docs/index.md:117]:
```suggestion
  be written to. Can also be sourced from the `EC_VERBOSE_FILE` environment variable.
  Defaults to `request.log`.
```

**Comment by sakurai-youhei** [comment]
@tobio Thanks for the point! Maybe moving the environment variable sentence before the default value sentence would also let the original `sourced from` be more accurate like the suggestions I prepared above?

**Comment by tobio** [comment]
Without reading the code, I'd expect the environment variable to override the value declared in the provider declaration. Perhaps we could have a section below the parameter descriptions which plainly states how these different sources override each other. Something like:

> ** Tip: Arguments specified in the module file take precedence over environment variables

**Review by tobio — APPROVED** [review]
LGTM, @elastic/platform-writers can someone take another quick look at the changes here too?

**Review by alaudazzi — APPROVED** [review]
LGTM for the text

**Pull request merged**